athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -0,0 +1,320 @@
+import com.amazonaws.athena.connectors.gcs.common.PartitionFolder;
+import com.amazonaws.athena.connectors.gcs.common.PartitionResult;
+import com.amazonaws.athena.connectors.gcs.common.StoragePartition;
+import com.amazonaws.athena.connectors.gcs.glue.GenericPartitionResolver;
+import com.amazonaws.athena.connectors.gcs.glue.GlueUtil;
+import com.amazonaws.athena.connectors.gcs.storage.StorageMetadata;
+import com.amazonaws.athena.connectors.gcs.storage.StorageSplit;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.Column;
+import com.amazonaws.services.glue.model.Database;
+import com.amazonaws.services.glue.model.Table;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import org.apache.arrow.util.VisibleForTesting;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.amazonaws.athena.connector.lambda.metadata.ListTablesRequest.UNLIMITED_PAGE_SIZE_VALUE;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.CLASSIFICATION_GLUE_TABLE_PARAM;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GCS_CREDENTIAL_KEYS_ENV_VAR;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GCS_SECRET_KEY_ENV_VAR;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.STORAGE_SPLIT_JSON;
+import static com.amazonaws.athena.connectors.gcs.GcsSchemaUtils.buildTableSchema;
+import static com.amazonaws.athena.connectors.gcs.GcsUtil.getGcsCredentialJsonString;
+import static com.amazonaws.athena.connectors.gcs.GcsUtil.splitAsJson;
+import static com.amazonaws.athena.connectors.gcs.storage.datasource.StorageDatasourceFactory.createDatasource;
+
+public class GcsMetadataHandler
+        extends GlueMetadataHandler
+{
+    public static final String TABLE_FILTER_IDENTIFIER = "gs://";
+    private static final Logger LOGGER = LoggerFactory.getLogger(GcsMetadataHandler.class);
+    /**
+     * used to aid in debugging. Athena will use this name in conjunction with your catalog id
+     * to correlate relevant query errors.
+     */
+    private static final String SOURCE_TYPE = "gcs";
+    // Env variable name used to indicate that we want to disable the use of Glue DataCatalog for supplemental
+    // metadata and instead rely solely on the connector's schema inference capabilities.
+    private static final String GLUE_ENV = "disable_glue";
+    private static final CharSequence GCS_FLAG = "gcs";
```

Comment:
This flag should be a lot more explicit, like:
`"google-cloud-storage-flag"`

There can be other Glue Databases that are not meant to be used with the google cloud storage connector with "gcs" substring in the location, like:
`"s3://somebucket/somethinggcssomething/something/"`

Reply:
Refactored with this suggestion.